src/lib.rs

```diff
@@ -250,6 +250,27 @@ impl From<Type> for Value {
     }
 }
 
+impl Type {
+    pub fn fits(&self, other: &Self) -> bool {
+        match (self, other) {
```

Comment:
Unsure about the naming here. How can two types fit?

Reply:
`assignable`?

Reply:
I agree that the naming is weird. I needed a name the communicates the asymmetry. I kinda want to use the "<=" operator, since that's how it works in my head, but it's not an exact match.

So it works like this:
```
Union(Int, Bool) fits Int
Int NOT fits Union(Int, Bool) 
```

I like the idea of assignable, but I read `assignable` as `mutable` or `modifyable`. I feel it brings the wrong connotations.

Maybe: `sub_type`, `contains`, `intersects`?